src/app/new-page/component/content-repeater/content-repeater.component.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  Component,
+  Input,
+  OnChanges,
+  QueryList,
+  SimpleChanges,
+  ViewChildren
+} from '@angular/core';
+import { ContentInputNewComponent } from '../content-input/content-input.component';
+import {
+  ContentItems,
+  Content,
+  ContentParser,
+  ContentItemsType
+} from 'src/app/services/xml-parser-service/xmp-parser.service';
+@Component({
+  selector: 'app-content-new-repeater',
+  templateUrl: './content-repeater.component.html',
+  styleUrls: ['./content-repeater.component.css']
+})
+export class ContentRepeaterNewComponent implements OnChanges {
+  @Input() items: Content[];
+  @ViewChildren(ContentInputNewComponent)
+  components: QueryList<ContentInputNewComponent>;
+
+  ready: boolean;
+  content: ContentItemsType[];
+
+  constructor() {}
+
+  ngOnChanges(changes: SimpleChanges) {
+    for (const propName in changes) {
+      if (changes.hasOwnProperty(propName)) {
+        switch (propName) {
+          case 'items': {
+            if (
+              !changes['items'].previousValue ||
+              changes['items'].currentValue !== changes['items'].previousValue
+            ) {
+              this.ready = false;
+              this.content = [];
+              this.init();
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private init(): void {
+    if (this.items?.length) {
+      this.items.forEach((content) => {
+        if (content['content']) {
+          const hasInputChildren = content['content'].filter(
+            (c) => ContentParser(c) === 'input'
+          );
+          if (hasInputChildren?.length) {
+            this.content.push({ type: 'form', content: content['content'] });
```

Comment:
I'm curious about this, what's the need of pushing a form content element here?

Reply:
From memory, it was because `content['content']` would return a type `Content[]` and not `Form`. So to check if it's a form, I check if there is an Input as a child.

Reply:
hmm, so technically inputs don't need to be direct children of the form in the content model, although I believe currently they are always direct children.